doc/sfp-cmis/CMIS-custom-SI-settings.md

```diff
@@ -0,0 +1,267 @@
+  2.3. Generate module key based on Data Path(DP) lane speed + module vendor name, + module vendor part number.    
+  2.4. Using this module key (2.3), the search begins for the detected port in the GLOBAL_MEDIA_SETTINGS section of the applicable direction (TX/RX) setting block. If the module key matches any entries or any search that matches for (2.5) to (2.8) sections, then SI attributes from this section are copied to the SI param attribute list (2.9).   
+  2.5. If no match happens in 2.4, reduce the search to module default + speed of the detected port in GLOBAL_MEDIA_SETTINGS  
+  2.6. If no match happens in 2.5, reduce the search to the speed of the detected port in GLOBAL_MEDIA_SETTINGS  
+  2.7. If no match happens in the GLOBAL_MEDIA_SETTINGS block, the search now begins in the PORT_MEDIA_SETTINGS block for the detected port. If no match happens in the PORT_MEDIA_SETTINGS block, the final search for the default block is done.  
+  2.8. If no match happens to the default block, then an empty attribute list (2.10) is returned.
+
+3. Get the attribute list and validate if the list is not empty, then proceed to process the SI setting param list. If the list is empty, continue with the AP_CONF (applying app code, EC = 0) and DP_INIT state in the CMIS state machine.
+
+4. Apply the application code to the configuration with Explicit Control (EC) = 0, and commit to the activate state.  
+Reference Register: Upper Page 10h bytes 145 –152 (desired ApSel Code)
+
+5. Read and cache the default or active TX/RX SI settings. 
+Reference Register: Upper Page 11h bytes 214 to 234
+
+6. Update the new values from the attribute list (3) to the cached SI list (5).
+  
+7. Write new EQ settings to Staged Control Set 0. EQ settings include: new SI attribute list from (6), disabling adaptive TX input EQ settings if applicable. Apply application code to config with EC = 1 and commit to the active set.  
+Reference Register: Upper Page 10h bytes 143 (Applying APSel Config using ApplyDPInit, Copying from Staged Control Set to Active Control Set).
+Reference Register: Upper Page 10h bytes 145 –152 (desired ApSel Code).   
+Reference Register: Upper page 10h byte 153-173 (desired Host defined SI settings for EC=1 mode)
+
+8. Validate the config_status code, if the status is not config success, then force CMIS to reinit and retry. If the configuration fails after 3 retry attempts, print an error message and exit from initializing this port. If config_status is successful, then continue with the DP_INIT state in the CMIS state machine.  
+Reference Register: Upper Page 11h Byte 202-205 (config_status register)
```

Comment:
@AnoopKamath even if module declares the config is success, Xcvrd should validate the "to be configured" SI settings with that in Active control set.

Reply:
I will take care of this. Thanks